lib/client/types/struct.go

```diff
@@ -1,70 +1,66 @@
-package client
+package types
```

Comment:
Hinted elsewhere, but I don't think this is a good idea. If you're sharing on-the-wire deserialization structs between packages, you're probably sharing too much information.

Besides, what if the JSON returned by Okta changes? Then you'll need to update the MFA code to deal with that, instead of the client

Reply:
I understand where you're coming from but I think it is unavoidable if we want to have the interface be limited and allow MFA device implementations to be fully featured in the future. One example that passing the entire factor object allows is the ability for the sms MFA device to support re-requesting a code with MFA action `verify/resend`.

The MFA implementation are heavily reliant on the Okta API spec, if it changes then the MFA implementations will need to be updated. We're currently using the `v1` okta API so all device implementations must use that API.